### ISSUE TITLE
Emojis don't appear on macOS

### ISSUE THREAD

**Issue by pauloewerton** [zim-desktop-wiki/zim-desktop-wiki#1862]
I'm on Zim v0.74.3_9 for macOS and emojis don't appear on the pages. I'm able to access the emoji navigator (Right click -> Insert Emoji), but they all appear as empty images.

**Comment by jaap-karssenberg** [comment]
This is not zim specific, but generic Gtk feature

**Comment by dehesselle** [comment]
I am aware and can reproduce, but I haven't investigated it yet.

**Comment by dehesselle** [comment]
Investigation shows that this is a long-standing upstream issue with Cairo. Originally this was an issue with both Pango and Cairo, but Pango has been fixed 3 years ago while the corresponding [merge request for Cairo](https://gitlab.freedesktop.org/cairo/cairo/-/merge_requests/9) is still open. It appears it is soon going to be superseded by some [recent work](https://gitlab.freedesktop.org/cairo/cairo/-/merge_requests/289) coming from gtk-osx's maintainer, which is a good sign.

Until that happens, I'm going to integrate that patch for Cairo on our end. Here's a first test:
<img width="416" alt="emoji_zim" src="https://user-images.githubusercontent.com/11375494/158040511-0fdacf4e-cd02-46df-b1ee-dc793e505b59.png">

**Comment by dehesselle** [comment]
This is an unsigned development build (Zim 0.74.3 build 53):
https://gitlab.com/dehesselle/zim_macos/-/jobs/2196401785/artifacts/raw/Zim.dmg

Can you confirm this is working @pauloewerton ?

**Comment by dehesselle** [comment]
@jaap-karssenberg Out of curiosity, does inserting emoji in a code block (gtksourceview) crash for you as well?

**Comment by jaap-karssenberg** [comment]
No, testing on linux (ubuntu with gnome) inserting an emoji in a code block
does not crash the application. Looks like it is just an unicode character
insertion. If I open the source file with GEdit, it even shows the emoji in
the source of the file - so looks like emoji is supported in basic unicode
font used on my system. From point of view of zim application code, there
is nothing special about emoji.

**Comment by dehesselle** [comment]
Thank you for testing. That's another macOS specific upstream issue then. (It also crashes without that patch applied.) I won't be able to resolve that.
